http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/RunnableRequest.java

```diff
@@ -74,34 +78,40 @@ public void run() {
      */
     private void initializePerRequestHandlers() {
         // Remove any existing handlers from the pipeline from the previous request.
-        removeIfExists(HttpStreamsClientHandler.class);
-        removeIfExists(ResponseHandler.class);
+        removeIfExists(HttpStreamsClientHandler.class, ResponseHandler.class,
+                       ReadTimeoutHandler.class, WriteTimeoutHandler.class);
 
+        // TODO make these configurable
```

Comment:
Can you modify this TODO to only add the write here, remove it when we finished writing, add the read then, then remove read when the request if finished?

Reply:
+1 done.